src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java

```diff
@@ -491,7 +491,7 @@ protected final void verify(Checker checker,
      */
     public final void verifyWithGoogleConfigParser(String[] listOfModules,
```

Comment:
Same here.

Reply:
Same with `getMasterGoogleConfig`, `getGoogleConfigChildren`, and anything else that says "google" in this abstract IT class.

Reply:
It looks like some methods could just be renamed and not split apart.

`getGoogleConfigChildren` really has no code specific to google, except in method name and one parameter name. So it can be renamed.

`getMasterGoogleConfig` looks like it could be made abstract and renamed to `getItConfig`, and the method in `AbstractGoogleModuleTestSupport` could just return `CONFIGURATION` which it already loads and caches for you.

Reply:
> getGoogleConfigChildren really has no code specific to google, except in method name and one parameter name. So it can be renamed.

This method is only used inside [verifyWithGoogleConfigParser()](https://github.com/checkstyle/checkstyle/blob/c3f84d08cce5ad488d54a0ca210bfcb0b475bb69/src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java#L163C23-L163C51) for creating the google config. So instead of renaming it, I have moved it to `AbstractGoogleModuleTestSupport` with other methods.

> getMasterGoogleConfig looks like it could be made abstract and renamed to getItConfig, and the method in AbstractGoogleModuleTestSupport could just return CONFIGURATION which it already loads and caches for you.

In the new changes, I created the `verifyWithItConfig()` method which will directly receive the configuration from `AbstractGoogleModuleTestSupport` as its parameter. So we will not need `getItConfig` to get the configuration.

Reply:
Here's the new code for you to understand it better:

https://github.com/checkstyle/checkstyle/blob/c3f84d08cce5ad488d54a0ca210bfcb0b475bb69/src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java#L138-L210

https://github.com/checkstyle/checkstyle/blob/c3f84d08cce5ad488d54a0ca210bfcb0b475bb69/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L458-L471

also take a look at this comment https://github.com/checkstyle/checkstyle/pull/15028#discussion_r1646095246, here I have explained the reason behind why I kept working of `verifyWithItConfig` like this.